src/videojs-hls.js

```diff
@@ -785,9 +815,10 @@ videojs.Hls.prototype.fillBuffer = function(offset) {
     return;
   }
 
+  // To determine how much is buffered, we need to find the buffered region we
+  // are currently playing in and measure it's length
   if (buffered && buffered.length) {
-    // assuming a single, contiguous buffer region
-    bufferedTime = tech.buffered().end(0) - tech.currentTime();
+    bufferedTime = Math.max(0, buffered.end(0) - currentTime);
```

Comment:
We can't assume a single region any more so we have to be smarter about calculating the amount of buffered data.